### PR TITLE
structured migration options

### PR DESCRIPTION
**Pull request by ruslandoga**
This PR attempts to add structured migration options as proposed in https://github.com/plausible/ecto_ch/pull/115#issuecomment-1716040253

### CODE REVIEW

File: test/ecto/adapters/clickhouse/connection_test.exs [hunk 1]
```diff
@@ -2065,6 +2102,10 @@ defmodule Ecto.Adapters.ClickHouse.ConnectionTest do
            ]
   end
 
+  # TODO?
+  @tag :skip
+  test "drop constraint on cluster"
```
Comment (ruslandoga) [inline test/ecto/adapters/clickhouse/connection_test.exs:2107]: https://hexdocs.pm/ecto_sql/Ecto.Migration.Constraint.html doesn't have `:options` field. We can hack around and pass cluster in `:comment` though :)

Comment (ruslandoga) [reply test/ecto/adapters/clickhouse/connection_test.exs:2107]: We can also open a PR into Ecto to add `:options`

Comment (edwardsmit) [reply test/ecto/adapters/clickhouse/connection_test.exs:2107]: I'm relatively new to ClickHouse, so maybe I'm just not fully aware, but according to the docs, constraints are created and dropped using `ALTER TABLE` statements https://clickhouse.com/docs/en/sql-reference/statements/alter/constraint

File: README.md [hunk 2]
```diff
@@ -94,6 +94,30 @@ MyApp.Repo.insert_all(MyApp.Example, rows, settings: [async_insert: 1])
...
 ```
 
+#### Migrations
+
+ClickHouse-specific options can be passed into `table.options` and `index.options`
+
+```elixir
+table_options = [cluster: "my-cluster"]
+engine_options = [order_by: "tuple()"]
+optiosn = table_options ++ engine_options
```
Comment (edwardsmit) [inline README.md:104]: `options` (spelling)

**Comment by edwardsmit** [comment]
@ruslandoga this looks great!

File: test/ecto/adapters/clickhouse/connection_test.exs (hunk 1 above)
Comment (ruslandoga) [reply test/ecto/adapters/clickhouse/connection_test.exs:2107]: Yes, but that's not the issue. The issue is that we we can't pass any `:options` to https://hexdocs.pm/ecto_sql/Ecto.Migration.html#constraint/3

File: README.md (hunk 2 above)
Comment (ruslandoga) [reply README.md:104]: 😅

**Pull request merged**